sdk/inc/azure/core/az_mqtt5_telemetry_consumer_codec.h

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition of #az_mqtt5_telemetry_consumer_codec.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_MQTT5_TELEMETRY_CONSUMER_CODEC_H
+#define _az_MQTT5_TELEMETRY_CONSUMER_CODEC_H
+
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief MQTT5 Telemetry Consumer Codec options.
+ *
+ */
+typedef struct
+{
+  /**
+   * @brief The service group id to use for the subscription topic, specifying this allows the
+   * telemetry consumer to subscribe to a shared subscription topic. It can be AZ_SPAN_EMPTY if not
+   * using a shared subscription topic.
+   */
+  az_span service_group_id;
+
+  /**
+   * @brief The topic format to use for the subscription topic to receive telemetry.
+   *
+   * @note Can include {name} for telemetry name, {serviceId} for model id, and/or {senderId} for
+   * the sender's client_id. The default value is services/{serviceId}/{senderId}/telemetry.
+   *
+   * @note Tokens must be surrounded by slashes, unless they are at the beginning or end of the
+   * topic.
+   *
+   */
+  az_span telemetry_topic_format;
+} az_mqtt5_telemetry_consumer_codec_options;
+
+/**
+ * @brief MQTT5 Telemetry Consumer Codec.
+ *
+ */
+typedef struct
+{
+  struct
+  {
+    az_span model_id;
+    az_span sender_id;
+
```

Comment:
nit: extra space.

Reply:
Also, missing documentation?

Reply:
Since both fields are internal I did not add documentation, I also ran the style checker so the code should be correctly indented